### PR TITLE
Fixes #35921 - Unify use of AccessLog

### PR DESCRIPTION
**Pull request by adamruzicka**
across http and https applications

### CODE REVIEW

**Comment by adamruzicka** [comment]
Should we unify on on access log being turned off or on? Alternatively we could do it conditionally based on logging configuration

File: lib/launcher.rb [hunk 1]
```diff
@@ -106,6 +106,7 @@ def https_app(https_port, plugins = https_plugins)
         :SSLOptions => ssl_options,
         :SSLCiphers => CIPHERS - Proxy::SETTINGS.ssl_disabled_ciphers,
         :daemonize => false,
+        :AccessLog => [],
```
Comment (ekohl) [inline lib/launcher.rb]: It took me a moment to understand how this was to unify it. Would it make sense to have a `base_app_settings` that returns all common options between HTTP and HTTPS? Things like `Logger`, `ServerSoftware` etc should be there and then merged with HTTP(S) specific options (`Port`, SSL settings).

Comment (adamruzicka) [reply lib/launcher.rb]: I wanted to make the changes as small as possible, but sure, done

**Review by ekohl — COMMENTED** [review]
If you want to make it clear, you can introduce a commit that introduces `base_app_settings` without `AccessLog` (as `Refs #35921`) and then one that sets it in base_app_settings.

I also wonder about a clear description for those who don't know what `AccessLog` does. Perhaps it should say something like "Prior to this change http access logs weren't logged while https logs were. After this, neither logs access logs." At least, that's if I properly understood the change. Did I?

**Comment by adamruzicka** [comment]
Ping

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**